lib/service/Service.ts

```diff
@@ -678,38 +682,64 @@ class Service {
       });
     }
 
-    this.orderBook.on('peerOrder.incoming', order => callback(order));
-    this.orderBook.on('ownOrder.added', order => callback(order));
+    this.orderBook.on('peerOrder.incoming', callback);
+    this.orderBook.on('ownOrder.added', callback);
+
+    const removalCallback = (orderRemoval: OrderPortion) => callback(undefined, orderRemoval);
+    this.orderBook.on('peerOrder.invalidation', removalCallback);
```

Comment:
These can also be merged together into 1 event since the callback is the same.

```
const orderRemoved$ = merge(
...
).pipe(
  takeUntil(cancelled$),
);
```